airbyte-webapp/src/core/form/schemaToFormBlock.ts

```diff
@@ -119,6 +119,15 @@ export const jsonSchemaToFormBlock = (
     };
   }
 
+  const type = (Array.isArray(jsonSchema.type) ? jsonSchema.type[0] : jsonSchema.type) ?? "null";
```

Comment:
It looks like here we are just grabbing the first value of `type` if it is an array, but on line 128 we reject `items` if its type is an array. 

Is there a reason the two shouldn't be consistent? If no then maybe we should grab the first value from `jsonSchema.items.type` if it is an array as well?

Reply:
The fact that we grab the first one for the top level type is a leftover because we currently have a few specs which set the type like this: `type: ["string", "null"]`. I would like to get rid of that eventually, but it requires a bunch of cleanup first.

For arrays we don't allow this by the acceptance test criteria:

Special case for type + null:
https://github.com/airbytehq/airbyte/blob/d82e01a34fe5d637fbdda394f5cad4ab11a78ffb/airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py#L290

Always fail if this is done in arrays:
 https://github.com/airbytehq/airbyte/blob/d82e01a34fe5d637fbdda394f5cad4ab11a78ffb/airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py#L336

Reply:
Ah that makes sense, I think this logic is good then